packages/swingset-liveslots/test/test-cache.js

```diff
@@ -0,0 +1,120 @@
+import test from 'ava';
+import '@endo/init/debug.js';
+
+import { makeCache } from '../src/cache.js';
+
+test('cache', t => {
+  const backing = new Map();
+  backing.set('key0', 'value0');
+
+  const log = [];
+  const readBacking = key => {
+    log.push(['read', key]);
+    return backing.get(key);
+  };
+  const writeBacking = (key, value) => {
+    log.push(['write', key, value]);
+    backing.set(key, value);
+  };
+  const deleteBacking = key => {
+    log.push(['delete', key]);
+    backing.delete(key);
+  };
+
+  const c = makeCache(readBacking, writeBacking, deleteBacking);
+  c.insistClear();
+
+  // new reads pass through immediately to backing store
+  t.is(c.get('key0'), 'value0');
+  t.is(c.get('key2'), undefined);
+  t.deepEqual(log, [
+    ['read', 'key0'],
+    ['read', 'key2'],
+  ]);
+  log.length = 0;
```

Comment:
Potential improvement:
```suggestion
  // new reads pass through immediately to backing store
  t.is(c.get('key0'), 'value0');
  t.deepEqual(log.splice(0), [['read', 'key0']]);
  t.is(c.get('key2'), undefined);
  t.deepEqual(log.splice(0), [['read', 'key2']]);
```
or abstracting into a helper like `const clearAndAssertLog = (t, log, expected) => t.deepEqual(log.splice(0), expected)`:
```suggestion
  // new reads pass through immediately to backing store
  t.is(c.get('key0'), 'value0');
  clearAndAssertLog(t, log, [['read', 'key0']]);
  t.is(c.get('key2'), undefined);
  clearAndAssertLog(t, log, [['read', 'key2']]);
```

Reply:
I usually do `t.deepEqual(log.shift(), ..)`, and yeah when that has felt verbose I've thrown together a little helper like that.

Any opinions on `log.shift()` vs `log.splice(0)`? The latter takes me a few seconds to remember how `splice` works.

Reply:
I like `splice` for supporting the ability to make assertions about a batch of entries rather than one at a time, and really this should be an external test helper where the implementation is explained inline but never really considered at call sites because it Just Works as expected (above suggestion updated with that in mind).
```js
/**
 * Clear the contents of a log array and assert that they match expectations.
 *
 * @param {import('ava').ExecutionContext} t
 * @param {unknown[]} log
 * @param {unknown[]} expected
 * @returns {void}
 */
const clearAndAssertLog = (t, log, expected) => {
  // `log.splice(0)` removes elements starting at index 0 (i.e., every element)
  // and returns them in an array.
  const entries = log.splice(0);
  t.deepEqual(entries, expected);
}
```